src/app/pages/registration/registration-page.component.ts

```diff
@@ -79,9 +68,17 @@ export class RegistrationPageComponent implements OnInit {
     return this.form.invalid && this.submitted;
   }
 
-  processParamMap(paramMap: ParamMap) {
-    return paramMap.keys
-      .filter(key => !['sso', 'userid'].includes(key))
-      .reduce((acc, key) => ({ ...acc, [key]: paramMap.get(key) }), { ...this.model });
+  extractModel(route: ActivatedRouteSnapshot) {
+    return Object.keys(route.queryParams)
+      .filter(key => !['userid'].includes(key))
+      .reduce((acc, key) => ({ ...acc, [key]: route.queryParams[key] }), { ...this.model });
+  }
+
+  extractConfig(route: ActivatedRouteSnapshot) {
+    return {
+      sso: !!route.url.find(segment => segment.path.includes('sso')),
+      userId: route.queryParams.userid,
```

Comment:
userid query param is now mandatory for all routes that include `sso`, am I right? Should this be documented?